lib/client.c

```diff
@@ -1965,6 +1969,7 @@ SF_STATUS STDCALL _snowflake_execute_ex(SF_STMT *sfstmt,
                     "localLocation");
 
             } else {
+                // todo: Deadlock possible?
```

Comment:
I do not think it could cause deadlock here. But still ping here in case I forgot some use scenarios.

Reply:
@sfc-gh-smirzaei To make it super safe, we could add some logs here as check-point. In case it causes deadlock(interrupt before the lock is released)